From: Lin Yun Sheng <linyunsheng@huawei.com>
Date: Fri, 4 Aug 2017 17:24:59 +0800
Subject: net: hns: Fix for __udivdi3 compiler error
Git-commit: 967b2e2a76e380abdebe7f3c7ab17e8831accd21
Patch-mainline: v4.14-rc1
References: bsc#1068693

This patch fixes the __udivdi3 undefined error reported by
test robot.

Fixes: b8c17f708831 ("net: hns: Add self-adaptive interrupt coalesce support in hns driver")
Signed-off-by: Yunsheng Lin <linyunsheng@huawei.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/net/ethernet/hisilicon/hns/hns_enet.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/hisilicon/hns/hns_enet.c b/drivers/net/ethernet/hisilicon/hns/hns_enet.c
index 832f27792e3f..36520634c96a 100644
--- a/drivers/net/ethernet/hisilicon/hns/hns_enet.c
+++ b/drivers/net/ethernet/hisilicon/hns/hns_enet.c
@@ -847,7 +847,8 @@ static void hns_update_rx_rate(struct hnae_ring *ring)
 
 	total_bytes = ring->stats.rx_bytes - ring->coal_last_rx_bytes;
 	time_passed_ms = jiffies_to_msecs(jiffies - ring->coal_last_jiffies);
-	ring->coal_rx_rate = (total_bytes / time_passed_ms) >> 10;
+	do_div(total_bytes, time_passed_ms);
+	ring->coal_rx_rate = total_bytes >> 10;
 
 	ring->coal_last_rx_bytes = ring->stats.rx_bytes;
 	ring->coal_last_jiffies = jiffies;
-- 
2.14.2

